lib/bundler/resolver/spec_group.rb

```diff
@@ -5,10 +5,12 @@ class Resolver
     class SpecGroup
       include GemHelpers
 
+      attr_reader :all_specs
       attr_accessor :name, :version, :source
       attr_accessor :ignores_bundler_dependencies
 
       def initialize(all_specs)
+        @all_specs = all_specs
```

Comment:
This is used to create a copy of `SpecGroup` in `Resolver`.
We can add a copy method to `SpecGroup` as another approach. Which approach do you like?

Note that we can't use `#dup` because it copies all instance variables that include `@activated_platforms`. We want a copy that just have the same `all_specs` and `ignores_bundler_dependencies`.

Reply:
You mean extracting the logic from `search_for` to `SpecGroup` with something like:

```ruby
def copy_for(platform = nil)
 sg_ruby = dup
 sg_ruby.activated_platforms = []

  if sg_ruby.for?(Gem::Platform::RUBY)
    sg_ruby.activate_platform!(Gem::Platform::RUBY)
    sg_ruby.activate_platform!(platform) if platform
    sg_ruby
  end
end
```

I like it, seems more DRY.

Reply:
Thanks. I like the `copy_for` name. I'll use it.